server/discordbot.py

```diff
@@ -70,6 +70,54 @@ async def on_message(self, message):
         if message.channel != self.channel:
             return
 
+        if message.content == "!getareas":
+            msg = ""
+            number_players = int(self.server.player_count)
+            msg += f"**Clients in Areas**\n"
+            for hub in self.server.hub_manager.hubs:
```

Comment:
This doesn't check if the hub even allows usage of /getareas in it, look into https://github.com/Crystalwarrior/KFO-Server/blob/8f2f8d73150fdf352b2a95663ba11386689ccb5f/server/client_manager.py#L1601 instead

Also it might be worth to split send_areas_clients and send_hubs_clients so they just send_ooc the string returned by a new function, it would simplify your code here and allow the `getareas`/`gethubs` algorithms to be re-used like in here.

Reply:
This was my initial approach to making !getareas work over discord, but the problem is the function requires it to be called from a client object since it checks if the user is a mod/GM/blinded. I could see this would be best implemented if I could just have another function that returns the info string instead of sending it to OOC. 

I was working on a KFO-Server that didn't have /gethubs implemented so !gethubs was never really something I had in mind.